frontend/src/components/Playback/PlayerElement.vue

```diff
@@ -1,66 +1,180 @@
+ */
+function onCanPlay(): void {
+  if (playbackManager.currentlyPlayingMediaType === 'Video') {
+    playerElement.applyCurrentSubtitle();
+  }
+}
+
+/**
+ * Callback for when HLS.js gets an error
+ */
+function onHlsEror(event: Events.ERROR, data: ErrorData): void {
+  if (data.fatal) {
+    switch (data.type) {
+      case ErrorTypes.NETWORK_ERROR: {
+        // try to recover network error
+        console.log('fatal network error encountered, try to recover');
+        hls?.startLoad();
+        break;
+      }
+      case ErrorTypes.MEDIA_ERROR: {
+        console.log('fatal media error encountered, try to recover');
+        hls?.recoverMediaError();
+        break;
+      }
+      default: {
+        // cannot recover
+        hls?.destroy();
+        break;
+      }
+    }
+  }
+}
+
 watch(
   () => playbackManager.currentSourceUrl,
-  () => {
-    if (
-      mediaElementRef.value &&
-      playbackManager.currentSourceUrl &&
-      playbackManager.currentMediaSource?.SupportsDirectPlay &&
-      playbackManager.currentlyPlayingMediaType === 'Audio'
-    ) {
-      mediaElementRef.value.src = playbackManager.currentSourceUrl;
+  async () => {
+    if (playbackManager.currentSourceUrl) {
+      /**
+       * Wait for nextTick to have the DOM updated accordingly
+       */
+      await nextTick();
+
+      try {
+        if (!mediaElementRef.value) {
+          throw new Error('No media element found');
+        }
+
+        if (
+          (playbackManager.currentMediaSource?.SupportsDirectPlay &&
+            playbackManager.currentlyPlayingMediaType === 'Audio') ||
+          (mediaElementRef.value.canPlayType('application/vnd.apple.mpegurl') &&
+            playbackManager.currentlyPlayingMediaType === 'Video')
+        ) {
+          /**
+           * For the video case, Safari iOS doesn't support hls.js but supports native HLS
+           */
+          mediaElementRef.value.src = playbackManager.currentSourceUrl;
+          mediaElementRef.value.currentTime = playbackManager.currentTime;
+        } else if (
+          Hls.isSupported() &&
+          playbackManager.currentlyPlayingMediaType === 'Video'
+        ) {
+          if (!hls) {
+            hls = new Hls({
+              testBandwidth: false,
+              startPosition: playbackManager.currentTime
+            });
+            hls.attachMedia(mediaElementRef.value);
+          } else {
+            hls.config.startPosition = playbackManager.currentTime;
+          }
+
+          hls.on(Events.ERROR, onHlsEror);
+
+          hls.loadSource(playbackManager.currentSourceUrl);
+        } else {
+          throw new Error('No playable case');
+        }
+      } catch {
+        useSnackbar(t('errors.cantPlayItem'), 'error');
+        playbackManager.stop();
+      }
     } else {
-      useSnackbar(t('errors.cantPlayItem'), 'error');
-      playbackManager.stop();
+      if (hls) {
+        hls.destroy();
+        hls = undefined;
```

Comment:
Make this a function as said above and use it everywhere hls must be destroyed

Reply:
Done